src/test/java/games/strategy/internal/persistence/serializable/IntegerMapProxyAsProxyTest.java

```diff
@@ -0,0 +1,29 @@
+package games.strategy.internal.persistence.serializable;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import games.strategy.persistence.serializable.AbstractProxyTestCase;
+import games.strategy.persistence.serializable.ProxyFactory;
+import games.strategy.util.IntegerMap;
+
+public final class IntegerMapProxyAsProxyTest extends AbstractProxyTestCase<IntegerMap<?>> {
+  @SuppressWarnings("unchecked")
+  public IntegerMapProxyAsProxyTest() {
+    super((Class<IntegerMap<?>>) (Class<?>) IntegerMap.class);
```

Comment:
Indeed.

```java
super((Class<IntegerMap<?>>) IntegerMap.class);
```

produces the error: `Cannot cast from Class<IntegerMap> to Class<IntegerMap<?>>`.

I believe this is because Java generics are invariant, which means you can't cast directly from `MyType<A>` to `MyType<B>`.  So you play this trick whereby you effectively upcast `MyType<A>` to the "supertype" `MyType<?>`, followed by a downcast to `MyType<B>`.  Probably something out of the "let's make sure you really want to shoot yourself in the foot" compiler check. :smile: